File: 104.二叉树的最大深度.py
#
# @lc app=leetcode.cn id=104 lang=python3
#
# [104] 二叉树的最大深度
#

# @lc code=start
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def maxDepth(self, root: TreeNode) -> int:
        if root:
            return max(self.maxDepth(root.left),self.maxDepth(root.right)) + 1
        else:
            return 0
# @lc code=end

File: 104.二叉树的最大深度.java
/*
 * @lc app=leetcode.cn id=104 lang=java
 *
 * [104] 二叉树的最大深度
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    // int Max = 0;
    public int maxDepth(TreeNode root) {
        int res = 0;  // 表示当前深度
        if (root == null){
            return 0;
        }
        res++;
        if(root.left!=null){
            res = Math.max(res, 1 + maxDepth(root.left));
            // return 1;
        }
        if(root.right!=null){
            res = Math.max(res, 1 + maxDepth(root.right));
            // return 1;
        }
        return res;
    }
}
// @lc code=end

